torchinductor/lowering.py

```diff
@@ -149,9 +154,7 @@ def wrapped(*args, **kwargs):
         assert not any(isinstance(x, TensorBox) for x in kwargs.values())
 
         if type_promote and indices:
-            dtype = functools.reduce(
-                torch.promote_types, [args[i].get_dtype() for i in indices]
-            )
+            dtype = get_promoted_dtype(*[args[i] for i in indices])
```

Comment:
`args` are filtered to include `TensorBox` only at line 152, but we should include scalars as well when computing the promoted type.